File: day03/aoc.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func getSolutionPart1() int {
    return 1337
}

func getSolutionPart2() int {
    return 42
}

func parseInput(input string) ([]int, error) {
	var ints []int

	lines := strings.Split(strings.TrimSpace(input), "\n")

	for _, line := range lines {
		i, err := strconv.Atoi(line)
		if err != nil {
			return nil, err
		}

		ints = append(ints, i)
	}

	return ints, nil
}

func main() {
	part := os.Getenv("part")

	if part == "part2" {
		fmt.Println(getSolutionPart2())
	} else {
		fmt.Println(getSolutionPart1())
	}
}

File: day06/aoc.R
getSolutionPart1 <- function() {
    1337
}

getSolutionPart2 <- function() {
    42
}

part <- Sys.getenv("part")

if (part == "part1") {
    print (getSolutionPart1())
} else {
    print (getSolutionPart2())
}
